soc/riscv/riscv-ite/it8xxx2/Kconfig.soc

```diff
@@ -59,6 +59,24 @@ config SOC_IT8XXX2_GPIO_H7_DEFAULT_OUTPUT_LOW
 	  capability to pull it down. We can only set it as output low,
 	  so we enable output low for it at initialization to prevent leakage.
 
+config SOC_IT8XXX2_USBPD_PHY_PORT_COUNT
+	int "it8XXX2 USBPD physical port counts"
+	default 2
+	help
+	  ITE EC chip it81202 and it81302 both have embedded integrated
+	  pd module (support two usbpd ports), this is different from
+	  standalone TCPC.
+
+config SOC_USBPD_ITE_ACTIVE_PORT_COUNT
+	int "Number of ITE USB PD active ports"
+	default 0
+	help
+	  This sets the number of active USB Power Delivery (USB PD) ports
+	  in use on the ITE microcontroller. The active port usage should
+	  follow the order of ITE TCPC port index, ex. if we're active only
+	  one ITE USB PD port, then the port should be 0x3700 (port0 register
+	  base), instead of 0x3800 (port1 register base).
+
```

Comment:
I don't like that these SoC-specific Kconfig symbols are then used in the driver. I understand that currently only `it8xxx2` uses the `gpio_ite_it8xxx2` driver, but what if some other SoC starts using it as well? Moreover, this also looks like a configuration that could and probably should be put in the dts.

Reply:
I rename config SOC_IT8XXX2_USBPD_PHY_PORT_COUNT to  SOC_USBPD_ITE_PHY_PORT_COUNT for all ITE chip using, instead of ITE specific chip. I also re-define it in specific chip config (ex.it81202, it81302), since different chip may not support the same USBPD PHY port count.
SOC_USBPD_ITE_ACTIVE_PORT_COUNT default value is 0 , and we can re-define it In board defconfig if it's not 0.